### ISSUE TITLE
container_env on latest is still running with the slow gem env path

### ISSUE THREAD

**Issue by jrafanie** [ManageIQ/manageiq-pods#759]
Note, I'm not seeing [the removal of gem env path](https://github.com/ManageIQ/manageiq-rpm_build/pull/145/commits/242e1d423966c1ccf8f06a37b269f8d8b2a2da80) in any released manageiq-base-worker podified image, which is expected since it wasn't backported but even latest doesn't have it: 

```
% docker image list | grep "manageiq/manageiq-base"
manageiq/manageiq-base-worker                                                                          latest-morphy   c8ecc5bc89de   15 hours ago    2.01GB
manageiq/manageiq-base-worker                                                                          latest          c4d29cf656ef   25 hours ago    2.01GB
manageiq/manageiq-base-worker                                                                          jansa-4         4ffc2a5a71dd   2 months ago    2.3GB
manageiq/manageiq-base-worker                                                                          lasker-1        8433d2c48158   2 months ago    1.92GB
manageiq/manageiq-base-worker                                                                          kasparov-1      dbba0ce9e18b   8 months ago    2.17GB
manageiq/manageiq-base-worker                                                                          jansa-3         8443b4875e09   8 months ago    2.13GB
manageiq/manageiq-base-worker                                                                          jansa-2         bc9c96853a9c   11 months ago   2.09GB
```

I can see that even morphy branch has [gem env path](https://github.com/ManageIQ/manageiq-rpm_build/blob/morphy/rpm_spec/manageiq.spec.in#L54) but I was surprised that latest builds weren't pulling in master of rpm_build.

```
% docker run --rm -it docker.io/manageiq/manageiq-base-worker:latest /bin/bash
[root@89e898f3cc0e /]# cat /opt/manageiq/manageiq-gemset/enable
export APPLIANCE="true"
export BUNDLE_GEMFILE=/var/www/miq/vmdb/Gemfile
export GEM_HOME=/opt/manageiq/manageiq-gemset
export GEM_PATH=/opt/manageiq/manageiq-gemset:$(gem env path)
export PATH=/opt/manageiq/manageiq-gemset/bin:$PATH
```

I was going to close this but noticed that latest was still bad.  Maybe there's a good reason we're not using the master branch of rpm_build on latest but it still looked bad, especially when `update-ca-trust` was added in latest and isn't offset by the removal of `gem env path`.

```
% docker run --rm -it docker.io/manageiq/manageiq-base-worker:latest /bin/bash
[root@11ed9604dd92 /]# time source /var/www/miq/vmdb/container_env
real  0m0.504s

% docker run --rm -it docker.io/manageiq/manageiq-base-worker:latest-morphy /bin/bash
[root@cb76d085b6b0 /]# time source /var/www/miq/vmdb/container_env
real  0m0.281s

% docker run --rm -it docker.io/manageiq/manageiq-base-worker:lasker-1 /bin/bash
[root@b0e911f6ad44 /]# time source /var/www/miq/vmdb/container_env
real  0m0.282s

% docker run --rm -it docker.io/manageiq/manageiq-base-worker:jansa-2 /bin/bash
[root@ff6d1e25e61c /]# time source /var/www/miq/vmdb/container_env
real  0m0.383s

% docker run --rm -it docker.io/manageiq/manageiq-base-worker:jansa-3 /bin/bash
[root@223b9cb46763 /]# time source /var/www/miq/vmdb/container_env
real  0m0.382s

% docker run --rm -it docker.io/manageiq/manageiq-base-worker:jansa-4 /bin/bash
[root@e04633508271 /]# time source /var/www/miq/vmdb/container_env
real  0m0.377s
```

![image](https://user-images.githubusercontent.com/19339/135655561-1837930c-3d96-45a9-a5c9-59743ca889ca.png)

From docker.io/manageiq/manageiq-base-worker:latest...
[container_env_timings_latest.csv](https://github.com/ManageIQ/manageiq-pods/files/7268719/container_env_timings_latest.csv)
 

https://github.com/ManageIQ/manageiq-rpm_build/blob/master/rpm_spec/manageiq.spec.in#L54 is master and doesn't contain the `gem env path`

**Comment by jrafanie** [comment]
from https://github.com/ManageIQ/manageiq-pods/issues/695

**Comment by jrafanie** [comment]
We found that the yum repo file was incorrect for master/latest builds:

```
[root@5ba438de95cc /]# yum list installed | grep manageiq
manageiq-core.x86_64                          13.0.0-20210930000048.el8                           @manageiq-14-najdorf-nightly
manageiq-gemset.x86_64                        13.0.0-20210930000048.el8                           @manageiq-14-najdorf-nightly
manageiq-pods.x86_64                          13.0.0-20210930000048.el8                           @manageiq-14-najdorf-nightly
manageiq-release.noarch                       14.0-1.el8                                          @@commandline
manageiq-system.x86_64                        13.0.0-20210930000048.el8                           @manageiq-14-najdorf-nightly
python3-bambou.noarch                         3.1.1-1.el8                                         @manageiq-14-najdorf-noarch
python3-pylxca.noarch                         2.1.1-1.el8                                         @manageiq-14-najdorf-noarch
python3-vspk.noarch                           5.3.2-1.el8                                         @manageiq-14-najdorf-noarch
wmi.x86_64                                    1.3.14-2.el8                                        @manageiq-14-najdorf
[root@5ba438de95cc /]# cat /var/www/miq/vmdb/BUILD
20210930000048_105cad7
[root@5ba438de95cc /]# cat /var/www/miq/vmdb/VERSION
morphy-pre
```

```
[root@5ba438de95cc /]# grep -C 2 "baseurl" /etc/yum.repos.d/manageiq-14-najdorf.repo
[manageiq-14-najdorf]
name=ManageIQ 14 (Najdorf) Release- $basearch
baseurl=https://rpm.manageiq.org/release/14-najdorf/el$releasever/$basearch
enabled=1
gpgcheck=0
--
[manageiq-14-najdorf-noarch]
name=ManageIQ 14 (Najdorf) Release- noarch
baseurl=https://rpm.manageiq.org/release/14-najdorf/el$releasever/noarch
enabled=1
gpgcheck=0
--
[manageiq-14-najdorf-source]
name=ManageIQ 14 (Najdorf) Release- source
baseurl=https://rpm.manageiq.org/release/14-najdorf/el$releasever/src
enabled=0
gpgcheck=0
--
[manageiq-14-najdorf-nightly]
name=ManageIQ 14 (Najdorf) Nightly- $basearch
baseurl=https://rpm.manageiq.org/release/13-morphy-nightly/el$releasever/$basearch # <---- -wrong and enabled
enabled=1
gpgcheck=0
--
[manageiq-14-najdorf-nightly-noarch]
name=ManageIQ 14 (Najdorf) Nightly- noarch
baseurl=https://rpm.manageiq.org/release/13-morphy-nightly/el$releasever/noarch # <---- -wrong but disabled
enabled=0
gpgcheck=0
--
[manageiq-14-najdorf-nightly-source]
name=ManageIQ 14 (Najdorf) Nightly- source
baseurl=https://rpm.manageiq.org/release/13-morphy-nightly/el$releasever/src # <---- -wrong but disabled
enabled=0
gpgcheck=0
```

**Comment by jrafanie** [comment]
Fixed by https://github.com/ManageIQ/manageiq-pods/pull/758

**Comment by jrafanie** [comment]
@bdunne will #758 fix the `baseurl` for the `manageiq-14-najdorf-nightly-noarch` and `manageiq-14-najdorf-nightly-source` in addition to `manageiq-14-najdorf-nightly`?

**Comment by Fryguy** [comment]
yes, the fixes that went into 14.0-2 were these: ManageIQ/manageiq-rpm_build#189
